dlc-manager/src/manager.rs

```diff
@@ -925,15 +925,27 @@ where
             .accepted_contract
             .dlc_transactions
             .cets[contract.cet_index];
-        let broadcasted_txid = broadcasted_cet.txid();
-        let confirmations = self
-            .wallet
-            .get_transaction_confirmations(&broadcasted_txid)?;
+        let cet_txid = broadcasted_cet.txid();
+        let confirmations = self.wallet.get_transaction_confirmations(&cet_txid)?;
         if confirmations >= NB_CONFIRMATIONS {
             let closed_contract = ClosedContract {
-                signed_contract: contract.signed_contract.clone(),
                 attestations: contract.attestations.clone(),
-                cet_index: contract.cet_index,
+                cet_txid,
+                contract_id: contract.signed_contract.accepted_contract.get_contract_id(),
+                temporary_contract_id: contract
+                    .signed_contract
+                    .accepted_contract
+                    .offered_contract
+                    .id,
+                counter_party_id: contract
+                    .signed_contract
+                    .accepted_contract
+                    .offered_contract
+                    .counter_party,
+                pnl: contract
+                    .signed_contract
+                    .accepted_contract
+                    .compute_pnl(contract.cet_index),
```

Comment:
A big verbose here. curious of your thoughts

Reply:
Yeah I don't think you can do better. Tbh I'm not super happy with the way I structured contract. The goal was to avoid copying fields between structs and working around the lack of inheritance, but I think copying field might be better at the end (which is the route I took for the channels)